agent/engine/docker_client_test.go

```diff
@@ -363,28 +362,19 @@ func TestCreateContainerTimeout(t *testing.T) {
 	defer done()
 
 	warp := make(chan time.Time)
-	wait := &sync.WaitGroup{}
-	wait.Add(1)
 	config := docker.CreateContainerOptions{Config: &docker.Config{Memory: 100}, Name: "containerName"}
 	mockDocker.EXPECT().CreateContainer(gomock.Any()).Do(func(x interface{}) {
 		warp <- time.Now()
-		wait.Wait()
-		// Don't return, verify timeout happens
-		// TODO remove the MaxTimes by cancel the context passed to CreateContainer
-		// when issue #1212 is resolved
 	}).MaxTimes(1)
-	metadata := client.CreateContainer(config.Config, nil, config.Name, xContainerShortTimeout)
+	metadata := client.CreateContainer(context.TODO(), config.Config, nil, config.Name, xContainerShortTimeout)
```

Comment:
In all of the tests, where you have passed `context.TODO()` directly, please modify those to be something like this:

```golang
ctx, cancel := context.WithCancel(context.TODO())
defer cancel()

// pass ctx
```

Otherwise, we'll have go routine leaks because of uncancelled context in edge cases.

Reply:
OK, sure.